pcdscalc/be_lens_calcs.py

```diff
@@ -0,0 +1,826 @@
+"""Module for Beryllium Lens Calculations."""
+import logging
+import os
+import shutil
+from datetime import date
+from itertools import product
+
+import numpy as np
+import xraydb as xdb
+
+logger = logging.getLogger(__name__)
+
+# Set of Be lenses with thicknesses.
+LENS_RADII = [50e-6, 100e-6, 200e-6, 300e-6, 500e-6, 1000e-6, 1500e-6]
+# Constant for converting between FWHM and sigma of a Gaussian function.
+FWHM_SIGMA_CONVERSION = 2.35482004503
+# Constant for converting between wavelength and photon energy.
+WAVELENGTH_PHOTON = 1.2398
+
+
+def photon_to_wavelength(energy):
+    """
+    Find the wavelength in micrometers.
+
+    Use photon energy in electronvolts. The equation is approximately: λ[µm] =
+    1.2398 / E[eV]. The photon energy at 1 μm wavelength, the wavelength of
+    near infrared radiation, is approximately 1.2398 eV.
+
+    Usage:
+
+    .. code-block:: python
+
+        photon_to_wavelength(8)
+
+    Parameters
+    ----------
+    energy : number
+        Photon energy in electronvolts
+
+    Returns
+    -------
+    Wavelength in micrometers
+    """
+    return WAVELENGTH_PHOTON / energy
+
+
+def gaussian_sigma_to_fwhm(sigma):
+    """
+    Convert between FWHM and sigma of a Gaussian function.
+
+    FWHM = 2.35482004503 * sigma
+
+    https://brainder.org/2011/08/20/gaussian-kernels-convert-fwhm-to-sigma/
+
+    Usage:
+
+    .. code-block:: python
+
+        gaussian_sigma_to_fwhm(0.3)
+
+
+    Parameters
+    ----------
+    sigma : float
+
+    Returns
+    -------
+    FWHM -  Full Width at the Half Maximum
+    """
+    return FWHM_SIGMA_CONVERSION * sigma
+
+
+def gaussian_fwhm_to_sigma(fwhm):
+    """
+    Convert between FWHM and sigma of a Gaussian function.
+
+    sigma = FWHM / 2.35482004503
+
+    https://brainder.org/2011/08/20/gaussian-kernels-convert-fwhm-to-sigma/
+
+    Usage:
+
+    .. code-block:: python
+
+        gaussian_fwhm_to_sigma(0.023)
+
+    Parameters
+    ----------
+    fwhm : float
+        Full Width at the Half Maximum
+
+    Returns
+    -------
+    sigma
+    """
+    return fwhm / FWHM_SIGMA_CONVERSION
+
+
+# TODO: this is a test file... take care of it
+LENS_SET_FILE = os.path.dirname(
+    __file__) + '/tests/test_lens_sets/lens_set.npy'
+
+
+def get_lens_set(set_number_top_to_bot, filename=LENS_SET_FILE):
+    """
+    Get the lens set from the file provided.
+
+    Parameters
+    ----------
+    set_number_top_to_bot : int
+        The Be lens holders can take 3 different sets that we usually set
+        before experiments, this is to specify what number set.
+    filename : str
```

Comment:
```suggestion
    filename : str, optional
```
Any parameters with default values should be denoted as optional. I think there are a couple more in this file.

Reply:
Thanks a lot, added optional all (i think) default values in 0f33d1e